File: LeetCode/LeetCode/rm_elem.cc
#include<iostream>
#include<vector>
using namespace std;
class Solution{
    public:
        int removeElement(vector<int>&nums, int val) {
            int left = 0, right = nums.size();
            while(left < right) {
                if(nums[left] != val) {
                    ++left;
                }else {
                    swap(nums[left], nums[--right]);
                }
            }
            return right;
        }
};

int main(void)
{
    vector<int> v = {1,2,3,4,5};
    Solution().removeElement(v, 3);
    for(auto i : v) {
        cout << i;
    }
    cout <<endl;
}

File: LeetCode/LeetCode/rm_elem.go
package main

import "fmt"

func removeElement(nums []int, val int) int {
	i , j := 0, len(nums)-1
	for{
		for i<len(nums) && nums[i] != val {
			i++
		}

		for j >= 0 && nums[j] == val {
			j--
		}

		if i >= j {
			break
		}

		nums[i], nums[j] = nums[j], nums[i]
	}
	return i
} 

func main() {
	v := []int{1,2,3,4}
	fmt.Println(removeElement(v, 3))
	fmt.Println(v)
}
